fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/AgilityContext.java

```diff
@@ -286,6 +294,9 @@ public void commitNow() {
                     final long stamp = lock.writeLock();
 
                     try {
+                        if (preCommit != null) {
+                            preCommit.accept(currentContext);
+                        }
```

Comment:
I see this comment from @jjezra 
> Wouldn't the right place to implement this is in commitNow()? We wish to written timestamp to be as new as possible, as it only becomes effective after transaction commit.

I think that's probably right, but it does mean that we could do a bunch of IO just to find out that we lost the FileLock. Probably best to assume that is rare, and leave the code as you have it.

Also, note that `FDBRecordContext` has a concept of commit checks, so you could call `currentContext.addCommitCheck` when the context is opened, and it would check by the time it is committed.